File: listeners/examples/doorbot-listener-example.php
<?php

    require_once('../DoorbotListener.class.php');

    class ExampleListener extends DoorbotListener {

        function doorOpened ($serial, $name) {
            print "The door was opened by {$name}, card serial {$serial}.\n";
        }

        function unknownCard ($serial) {
            print "Unknown card {$serial} presented at the door.\n";
        }

        function doorbell () {
            print "Doorbell pressed.\n";
        }

        function startup () {
            print "Doorbot started.\n";
        }

    }

    $listener = new ExampleListener();
    $listener->listen();

File: listeners/examples/doorbot-listener-example.py
#!/usr/bin/env python
import socket, random, sys
sys.path.append('../')
import DoorbotListener

class ExampleListener(DoorbotListener.DoorbotListener):

    def startup(self):
        print "Doorbot started."

    def doorbell(self):
        print "Doorbell pressed."

    def doorOpened(self, serial, name):
        print "Door opened by %s, card serial %s." % (name, serial)

    def unknownCard(self, serial):
        print "Unknown card %s presented at the door." % (serial)

    def trigger(self, filename):
        print "Audio file [%s] triggered." % (filename)

listener = ExampleListener()
listener.listen()
